volumes/miovision/README.md

```diff
@@ -278,15 +299,86 @@ The tables below are produced using functions explained in the [API Puller](http
+
+1) Download the table `miovision_api.intersections` into a csv file and manually add the new intersections information into the table (coordinates, px, int_id, geom, which leg_restricted). 
+
+	a) New intersections name and details such as `intersection_uid`, `id`, `intersection_name` can be found using the [Miovision API](http://beta.docs.api.miovision.com/#!/Intersections/get_intersections). `date_installed` and `date_decommissioned` can be found by finding the first / last datetime_bin for that intersection_uid from that website. The last date for the old location can also be found from table `miovision_api.volumes`. 
+	
+	b) `px` for the intersection can then be found easily from this [web interface](https://demo.itscentral.ca/#).
+	
+	c) With the `px` information found above, get the rest of the information (such as `geom`, `lat`, `lng` and `int_id`) from table `gis.traffic_signal`.
+	
+	d) In order to find out which leg of that intersection is restricted, go to Google Map to find out the direction of traffic.
+
+2) Now that the updated table of `miovision_api.intersections` is ready, we have to update the table `miovision_api.intersection_movements`. We need to find out all valid movements for the new intersections from the data but we dont have that yet, so the following has to be done.
+
+	a) Fist, we have to create a couple of dummy tables (`volumes_test`, `volumes_15min_tmc_test`, `volumes_15min_test`) & sequences (`volumes_volume_uid_test_seq`, `volumes_15min_tmc_volume_15min_tmc_uid_test_seq`, `volumes_15min_volume_15min_uid_test_seq`) & function (`aggregate_15_min_tmc_get_valid_movement`).
+	
+	b) **Don't forget to grant `miovision_api_bot` access permission to all those new tables!**
+	
+	c) Modify the script `intersection_tmc.py` to only pull the new intersections into the dummy table and aggregate them into 15min tmc bins and **not** do the other things such as `report_date` etc.
+	
+	d) Once we have finish pulling and aggregating the data, run the below query. 
+	```sql
+	--table with all valid movements of new intersections
+	SELECT DISTINCT intersection_uid, classification_uid, leg, movement_uid
+		INTO miovision_api.intersection_movements_new
+		FROM miovision_api.volumes_15min_tmc_test;
+		ALTER TABLE miovision_api.intersection_movements_new ADD UNIQUE (intersection_uid, classification_uid, leg, movement_uid);
+		COMMENT ON TABLE miovision_api.intersection_movements_new IS 'Unique movements for each NEW intersection by classification';
+
+	--add the old ones into the old table (no need to remove decommissioned one cause what if we need them later on)
+	INSERT INTO miovision_api.intersection_movements_new
+	SELECT *
+	FROM miovision_api.intersection_movements
+	```
+
+3) Now that all the necessary source files are ready, it's time to test run everything! Create new function by directly copying from the old one but only change the table/function name it is reading from. Modify `intersection_tmc.py` again and run it.
```

Comment:
Why can't they just run the API script normally with the new intersection and then run `intersection_movements` from there? This seems very complicated.

Reply:
Cause the new intersections would not get inserted into `miovision_api.volumes_15min_tmc` because of these lines on function `miovision_api.aggregate_15_min_tmc()`. No bins will be created for the new intersections since they do not exist in the `intersection_movements` table just yet.

```sql
WITH zero_padding_movements AS (
...
FROM miovision_api.intersection_movements m
	CROSS JOIN generate_series(start_date - interval '1 hour', end_date - interval '1 hour 15 minutes', INTERVAL '15 minutes') AS dt(datetime_bin15)
.
.
FROM zero_padding_movements pad
LEFT JOIN --other tables before inserting into the 15min tmc table
```

Reply:
Couldn't they insert the new movements into the movement table by doing
```sql
INSERT INTO miovision_api.intersection_movements
SELECT DISTINCT intersection_uid, classification_uid, leg, movement_uid
	FROM miovision_api.volumes
        WHERE intersection_uid IN (SELECT intersection_uid FROM new_intersections
              AND datetime_bin > today() - interval '1 day'
```

Reply:
Trueeee, but that also means that we have to re-run other process such as `miovision_api.aggregate_15_min_tmc()` , `miovision_api.aggregate_15_min()` etc for the new ones then right?